app/bundles/LeadBundle/Event/LeadBuildSearchEvent.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+/*
+ * @copyright  2014 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Event;
+
+use Doctrine\DBAL\Query\QueryBuilder;
+use Doctrine\ORM\EntityManager;
+use Mautic\CoreBundle\Event\CommonEvent;
+
+/**
+ * Class LeadBuildSearchEvent.
+ */
+class LeadBuildSearchEvent extends CommonEvent
+{
+    /**
+     * @var string
+     */
+    protected $string;
+
+    /**
+     * @var QueryBuilder
+     */
+    protected $queryBuilder;
+
+    /**
+     * @var string
+     */
+    protected $alias;
+
+    /**
+     * @var string
+     */
+    protected $command;
+
+    /**
+     * @var string
+     */
+    protected $subQuery;
+
+    /**
+     * @var bool
+     */
+    protected $negate;
+
+    /**
+     * @var bool
+     */
+    protected $isSearchDone;
+
+    /**
+     * @var bool
+     */
+    protected $returnParameters;
+
+    /**
+     * @var bool
+     */
+    protected $strict;
+
+    /**
+     * @var array
+     */
+    protected $parameters;
+
+    /**
+     * @param string        $string
+     * @param string        $command
+     * @param string        $alias
+     * @param string        $negate
+     * @param QueryBuilder  $queryBuilder
+     * @param EntityManager $entityManager
+     */
+    public function __construct($string, $command, $alias, $negate, QueryBuilder $queryBuilder, EntityManager $entityManager)
```

Comment:
Why the event needs a EntityManager? Should not be here - if a subscriber needs it, it should be injected directly.